pkg/provider/azure_utils.go

```diff
@@ -604,3 +604,17 @@ func getServiceIPFamily(service *v1.Service) string {
 	}
 	return consts.IPVersionIPv4String
 }
+
+func getResourceGroupAndNameFromNICID(ipConfigurationID string) (string, string, error) {
```

Comment:
Could you add some comments for newly added functions?

Reply:
added